### PR TITLE
preserve version formatting as specified in desc

### LINKED ISSUE #1170: preserve formatting of package version in DESCRIPTION
E.g. `renv` currently has, in its DESCRIPTION file:

```
Version: 0.8.2-3
```

However, the pkgdown website shows in the top-left corner:

Screen Shot 2019-10-28 at 11 01 07 PM

Note that the `-` has been transformed to a `.`. It would be nice if `pkgdown` could preserve the original formatting.

### PR DESCRIPTION
**Pull request by kevinushey**
Closes #1170.

### CODE REVIEW

**Comment by jayhesselberth** [comment]
Can you add a test for this change to confirm that the version is now maintained verbatim?

**Comment by kevinushey** [comment]
```
> test_check("pkgdown")
── 1. Error: formatting in DESCRIPTION version is preserved (@test-build-version
cannot open the connection
Backtrace:
 1. pkgdown:::build_home_index(pkg, quiet = TRUE)
 2. pkgdown::render_page(pkg, "home", data, "index.html", quiet = quiet)
 3. pkgdown:::write_if_different(pkg, rendered, path, quiet = quiet)
 4. pkgdown:::write_lines(contents, path = full_path)
 5. base::writeLines(enc2utf8(text), path, useBytes = TRUE)
 6. base::file(con, "w")
```

For whatever reason, this works locally for me but not on Travis. :(

File: tests/testthat/test-build-version.R [hunk 1]
```diff
@@ -0,0 +1,9 @@
+context("test-build-version.R")
+
+test_that("formatting in DESCRIPTION version is preserved", {
+  pkg <- as_pkgdown(test_path("assets/version-formatting"))
+  expect_equal(pkg$version, "1.0.0-9000")
+  build_home_index(pkg, quiet = TRUE)
+  index <- read_lines(file.path(pkg$dst_path, "index.html"))
```
Comment (jayhesselberth) [inline tests/testthat/test-build-version.R]: Can you try replacing this with:

``` r
index <- read_lines(path(pkg$dst_path, "index.html"))
```

**Comment by hadley** [comment]
Can you please add a bullet to the top of `NEWS.md`? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

(The macOS-latest failure is expected)

**Comment by hadley** [comment]
Thanks @kevinushey!

**Pull request merged**